src/Guild.js

```diff
@@ -25,6 +25,29 @@ class Guild {
     autoBind(this);
   }
 
+  getRoleNameByID (value) {
+    return Object.keys(this.roles).find(key => this.roles[key] === value);
+  }
+
+  addRole (id, name) {
+    this.roles[name] = id;
+  }
+
+  updateRoleName (id, name) {
+    const oldKey = this.getRoleNameByID(id);
+
+    if(this.isRole(name)) {
+      return;
+    }
+
+    this.roles[name] = id;
+    delete this.roles[oldKey];
+  }
+
+  deleteRole(name) {
```

Comment:
My bad. Turns out the Delete event only includes Role id as `msg.d.role_id`. If you're happy to make this change, that'd be great. Otherwise, as it's my bad, I'm more than happy to make that fix after merging. Sorry.

Reply:
No problem, ill take a look at it now 👍

Reply:
Okay, i think i have the solution for this one, as we only pass in `msg.d.role_id`, it would be best to use `getRoleNameById` for getting the right key to delete.  

fixed this now.